### PR TITLE
[Keymap] rootiest for planck/rev6/

### PR DESCRIPTION
**Pull request by rootiest**
## Description

Rootiest keymap for planck/rev6

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/planck/keymaps/rootiest/config.h [hunk 1]
```diff
@@ -0,0 +1,134 @@
+#pragma once
```
Comment (drashna) [inline keyboards/planck/keymaps/rootiest/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/planck/keymaps/rootiest/config.h [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+/*
+ * USB/POWER OPTIONS
+ */
+#define USB_MAX_POWER_CONSUMPTION 500
```
Comment (drashna) [inline keyboards/planck/keymaps/rootiest/config.h:6]: This line doesn't need to be defined, as it's the default value. 
```suggestion
// #define USB_MAX_POWER_CONSUMPTION 500
```

File: keyboards/planck/keymaps/rootiest/keymap.c [hunk 3]
```diff
@@ -0,0 +1,1920 @@
...
+#define _SYMBLS 3
+#define _RAISE 3
+#define _NUMBRS 4
+#define _LOWER 4
+#define _PLOVER 5
+#define _FEATURS 6
+#define _ADJUST 6
+#define _NUMPD 7
+#define _TABULA 8
+#define _MOUSY 9
```
Comment (drashna) [inline keyboards/planck/keymaps/rootiest/keymap.c:34]: We highly recommend using enums for layer naming, as it's simpler to maintain, long term. 
```suggestion
enum custom_layers {
    _BASE,
    _QWERTY = _BASE,
    _COLEMAK,
    _DVORAK,
    _SYMBLS,
    _RAISE = _SYMBLS,
    _NUMBRS,
    _LOWER = _NUMBRS,
    _PLOVER,
    _FEATURS,
    _ADJUST = _FEATURS,
    _NUMPD,
    _TABULA,
    _MOUSY,
};
```

File: keyboards/planck/keymaps/rootiest/keymap.c [hunk 4]
```diff
@@ -0,0 +1,1920 @@
...
+#define UC_DEGF UC(0x2109)  // ℉
+
+// Time (ms) to wait between frames for the wake rgb animation
+#define WAKE_ANIMATION_TIMER_FREQUENCY 50
+
+// Is a display connected
+// #define USING_OLED_DISPLAY false
+
+// wait DELAY ms before unregistering media keys
+#define MEDIA_KEY_DELAY 10
```
Comment (drashna) [inline keyboards/planck/keymaps/rootiest/keymap.c:50]:
```suggestion
#define TAP_CODE_DELAY 10
```

File: keyboards/planck/keymaps/rootiest/keymap.c [hunk 5]
```diff
@@ -0,0 +1,1920 @@
...
+        uint8_t rgbcon_tracker : 8;
+        bool    do_wakeup_audio : 1;
+    };
+} user_config_t;
+
+user_config_t user_config;
+
+// Initialize variable holding the binary
+// representation of active modifiers.
+uint8_t mod_state;
```
Comment (drashna) [inline keyboards/planck/keymaps/rootiest/keymap.c:107]:
```suggestion
uint8_t mod_state = 0;
```

File: keyboards/planck/keymaps/rootiest/keymap.c [hunk 6]
```diff
@@ -0,0 +1,1920 @@
...
+        }
+    */
+}
+
+// EEPROM is getting reset!
+void eeconfig_init_user(void) {
+    user_config.raw                 = 0;
+    user_config.do_wakeup_animation = false;  // We want this disabled by default
+    user_config.do_wakeup_audio     = false;  // We want this disabled by default
+    user_config.rgbcon_tracker      = 0;
```
Comment (drashna) [inline keyboards/planck/keymaps/rootiest/keymap.c:1906]:
```suggestion
```

You don't need these if you're setting the raw to 0.  That will zero out all of the settings for the structure, so no need to explicitly re-zero them.

**Review by drashna — CHANGES_REQUESTED** [review]
Nice! 

Also:

File: keyboards/planck/keymaps/rootiest/config.h (hunk 1 above)
Comment (rootiest) [reply keyboards/planck/keymaps/rootiest/config.h:1]: Sure, will do!

Comment (drashna) [reply keyboards/planck/keymaps/rootiest/config.h:1]: Looks like it didn't get added/pushed

Comment (rootiest) [reply keyboards/planck/keymaps/rootiest/config.h:17]: Oh sorry, I did the keymap.c but not the config.h! Should be good now.

**Review by drashna — APPROVED** [review]

**Review by noroadsleft — APPROVED** [review]

File: keyboards/planck/keymaps/rootiest/config.h [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+    - MIDI notes can be added to the keymap
+    - Octave shift and transpose
+    - Virtual sustain, portamento, and modulation wheel
+    - etc.
+*/
+#define MIDI_ADVANCED
+
+/* override number of MIDI tone keycodes (each octave adds 12 keycodes and allocates 12 bytes) */
+//#define MIDI_TONE_KEYCODE_OCTAVES 2
+
```
Comment (fauxpark) [inline keyboards/planck/keymaps/rootiest/config.h:62]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Comment by rootiest** [comment]
Thank _**you**_!

I'm so happy my keymap was finally approved! 🥳

**Pull request merged**